util/encoding/encoding.go

```diff
@@ -599,3 +600,57 @@ func DecodeIfNull(b []byte) ([]byte, bool) {
 	}
 	return b, false
 }
+
+// EncodeTime encodes a time value, appends it to the supplied buffer,
+// and returns the final buffer. The encoding is guaranteed to be ordered
+// Such that if t1.Before(t2) then after EncodeTime(b1, t1), and
+// EncodeTime(b2, t1), Compare(b1, b2) < 0. The time zone offset is added
+// as a two byte encoding at the end of the time encoding.
+func EncodeTime(b []byte, t time.Time) []byte {
```

Comment:
I don't believe `Time.MarshalBinary` has any guarantees about the encoding. In particular, it doesn't guarantee any ordering of the encoded values. It just happens that the current encoding provides the ordering we need, but relying on that seems unwise.